Source/SwiftLintFramework/Helpers/BodyLineCounter.swift

```diff
@@ -0,0 +1,63 @@
+import SwiftSyntax
+
+/// Counts the number of lines in a scope's body.
+enum BodyLineCounter {
+    struct Result {
+        let exceeds: Bool
+        let lineCount: Int
+    }
+
+    /// This function determines if given a scope with a left/right brace, such as a function, closure, type, etc, if
+    /// the "body" of that scope meets or exceeds a specified line count limit when you ignore lines only containing
+    /// comments and/or whitespace.
+    ///
+    /// - parameter file:           The source file.
+    /// - parameter leftBraceLine:  The line where the scope's left opening brace is located.
+    /// - parameter rightBraceLine: The line where the scope's right closing brace is located.
+    /// - parameter limit:          The body line limit to stay under, ignoring lines only containing comments and/or
+    ///                             whitespace.
+    ///
+    /// - returns: A tuple containing whether the body `exceeds` the specific `limit` and `lineCount`, the number of
+    ///            effective lines of the body ignoring lines only containing comments and/or whitespace.
+    static func lineCountIgnoringCommentsAndWhitespace(
+        file: SwiftLintFile, leftBraceLine: Int, rightBraceLine: Int, limit: Int
+    ) -> Result {
```

Comment:
As the method name suggests it could just return the line count as an `Int` and the leave the assessment of it to the caller. This would also make the `limit` parameter unnecessary.

Reply:
Yes that would probably be better. Let me try that.